src/core/scenario.jl

```diff
@@ -80,6 +80,7 @@ All values of `d` are expected to be `Dicts` except for `PV` and `GHP`, which ca
     handle conversion of Vector of Vectors from JSON to a Matrix in Julia).
 """
 function Scenario(d::Dict; flex_hvac_from_json=false)
+    d = deepcopy(d)
```

Comment:
do we need this, and if so why?

Reply:
It's not absolutely necessary, but this function actually mutates the inputs dict `d`; this deepcopy avoids that mutation. I think it's cleaner in a workflow where you define that inputs_dict and expect it to be preserved for later use (e.g. running again with the same or slightly modified inputs_dict, or comparing the raw inputs with processed inputs).